athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/glue/GlueFieldLexer.java

```diff
@@ -89,19 +90,22 @@ private static Field lexComplex(String name, GlueTypeParser.Token startToken, Gl
                     + " but found " + startToken.getMarker());
         }
 
-        if (startToken.getValue().toLowerCase().equals(STRUCT)) {
+        final String startTokenValueLower = startToken.getValue().toLowerCase();
+
+        if (startTokenValueLower.equals(STRUCT)) {
             fieldBuilder = FieldBuilder.newBuilder(name, Types.MinorType.STRUCT.getType());
         }
-        else if (startToken.getValue().toLowerCase().equals(LIST)) {
+        else if (startTokenValueLower.equals(ARRAY) || startTokenValueLower.equals(SET)) {
```

Comment:
what happened to list?

Reply:
What do you mean?
LIST was misnamed before, see line 40 in athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/glue/GlueFieldLexer.java in the diff